docs/TpchBenchmark.md

```diff
@@ -0,0 +1,453 @@
+    <tr><td>cache_prefetch_min_pct</td><td>Best is 80</td><td>Best is 80</td></tr>
+    <tr>
+        <td colspan="3" style="font-size: 8pt">
+&nbsp;&nbsp;*&nbsp;<b>vCPUs</b> = (cores * hyper-threads)<br />
+**&nbsp;Wide tables and few columns retrieved per row can lead to many I/O requests, suggest increasing this value based on testing and the need to reduce small I/O requests.
+        </td>
+    </tr>
+</table>
+
+## Optimizations for the TpchBenchmark (Single Process Use Case)
+
+### ___num_drivers___
+This configuration option describes the number of resources available to process query plan tasks. This can greatly improve performance of CPU bound workloads. The recommendation in the table above seems to be optimal for TPCH-like workloads.
+
+### ___num_io_threads___
+This configuration option describes the number of threads available per driver to retrieve data from the network source. If the workload is I/O bound, then increasing this number is beneficial if the data is fewer requests of larger chunks as opposed to many smaller requests.
+
+### ___cache_gb___
+This configuration option is useful for workloads that read the same data several times per query but only applies to the single process use case. NOTE: There is a SSD Caching option in Velox but it to is ONLY useful in the single process use case.
+
+### ___num_splits_per_file___
+This configuration option is best when the data set count of row groups matches this value. The affect in overall performance appears based on testing to be small, however.
+
+## Optimizations for All Workloads (Both Use Cases)
+
+### ___max_coalesce_bytes___
+This configuration option is the maximum bytes coalesced into a single request to the data source. This was tested from the default 128MB to 2GB, and the overall improvement was small as size increased. Capturing request data did show larger and fewer requests but not enough to vastly improve I/O performance. 
+
+### ___max_coalesce_distance_bytes___
+This configuration option is the maximum byte distance between needed data in the same file at the data source that can be coalesced. Increasing this value would theoretically reduce the number of requests and increase each request size. However, if made too large the query will return too many un-needed bytes and could decrease I/O performance. This plus __max_coalesce_bytes__ should be fine-tuned for the workload being run.
+
+## Summary
+If a use of Velox matches the use case of the TcphBenchmark then it is a good tool to test, I/O and driver performance for specific TCP-H queries. This would benefit execution of specific production workloads that are like the chosen queries. If in multi-process use case, like Spark/Gluten/Velox configuration, the recommendation is to oversubscribe I/O threads between 2X and 3X vCPUs and tune the 2 coalesce configurations exposed.
+
+---
+## Appendix A: Patch to Build TpchBenchmark Executable
```

Comment:
see my comment above. We should provide an easy way for users to build this without manually tweaking the build system. Cc: @kgpai @assignUser

Reply:
Yes I agree, the options should be set via cli/make (or added via a var + if()). We can just add another target to the make file(tpch-io?)

I am not sure why s3 is only added via patch and not permanently. If it is not always needed and just for the io benchmark than please add an option and use a second `target_link_libraries` call to add it if needed.

Reply:
IIRC there was just a PR to bump the aws sdk version because of a build error. I can assists on the cmake changes feel free to ping me in the velox slack :)

Reply:
I now correctly have the TpchBencmark tool building with S3 support.  However, the tool does not actually support S3 (it is clearly using an s3:// file as a local filename. I have the diff that I pushed it as a PR draft (PR#5366) for folks to see I will post here when that is done. I thought Orri's changes to support S3 were merged into main but maybe they were not??

Reply:
https://github.com/facebookincubator/velox/pull/5366